tests/collections/asr/decoding/test_rnnt_decoding.py

```diff
@@ -0,0 +1,252 @@
+        decoded_hyps = decoding.decode_hypothesis(n_hyps)  # type: List[str]
+
+        hypotheses.append(decoded_hyps[0])  # best hypothesis
+        all_hypotheses.append(decoded_hyps)
+
+    return hypotheses, all_hypotheses
+
+
+class TestRNNTDecoding:
+    @pytest.mark.unit
+    def test_constructor(self):
+        cfg = RNNTDecodingConfig()
+        vocab = char_vocabulary()
+        decoder = get_rnnt_decoder(vocab_size=len(vocab))
+        joint = get_rnnt_joint(vocab_size=len(vocab))
+        decoding = RNNTDecoding(decoding_cfg=cfg, decoder=decoder, joint=joint, vocabulary=vocab)
+        assert decoding is not None
+
+    @pytest.mark.unit
+    def test_constructor_subword(self, tmp_tokenizer):
+        cfg = RNNTBPEDecodingConfig()
+        vocab = tmp_tokenizer.vocab
+        decoder = get_rnnt_decoder(vocab_size=len(vocab))
+        joint = get_rnnt_joint(vocab_size=len(vocab))
+        decoding = RNNTBPEDecoding(decoding_cfg=cfg, decoder=decoder, joint=joint, tokenizer=tmp_tokenizer)
+        assert decoding is not None
+
+    @pytest.mark.skipif(
+        not NUMBA_RNNT_LOSS_AVAILABLE, reason='RNNTLoss has not been compiled with appropriate numba version.',
+    )
+    @pytest.mark.with_downloads
+    @pytest.mark.unit
+    def test_greedy_decoding_preserve_alignments(self, test_data_dir):
+        model, encoded, encoded_len = get_model_encoder_output(test_data_dir, 'stt_en_conformer_transducer_small')
+
+        beam = greedy_decode.GreedyRNNTInfer(
+            model.decoder,
+            model.joint,
+            blank_index=model.joint.num_classes_with_blank - 1,
+            max_symbols_per_step=5,
+            preserve_alignments=True,
+        )
+
+        enc_out = encoded
+        enc_len = encoded_len
+
+        with torch.no_grad():
+            hyps = beam(encoder_output=enc_out, encoded_lengths=enc_len)[0]  # type: rnnt_utils.Hypothesis
+            hyp = decode_text_from_greedy_hypotheses(hyps, model.decoding)
+            hyp = hyp[0]
+
+            assert hyp.alignments is not None
+
+            # Use the following commented print statements to check
+            # the alignment of other algorithms compared to the default
+            # print("Text", hyp.text)
+            for t in range(len(hyp.alignments)):
+                t_u = []
+                for u in range(len(hyp.alignments[t])):
+                    logp, label = hyp.alignments[t][u]
+                    assert torch.is_tensor(logp)
+                    assert torch.is_tensor(label)
+
+                    t_u.append(int(label))
+
+            #     print(f"Tokens at timestep {t} = {t_u}")
+            # print()
+
+    @pytest.mark.skipif(
+        not NUMBA_RNNT_LOSS_AVAILABLE, reason='RNNTLoss has not been compiled with appropriate numba version.',
+    )
+    @pytest.mark.with_downloads
+    @pytest.mark.unit
+    @pytest.mark.parametrize(
+        "beam_config",
+        [
+            {"search_type": "greedy"},
```

Comment:
I don't really know the details about all these algorithms, but if any of them are expected to produce exactly the same label sequence, that would be a nice thing to test for as well. Or maybe something like greedy = beam with beam size of 1. Very different code paths but should give equivalent results - always a great test to have, in my opinion.

Reply:
It auto switches to greedy decoding at beam size 1 (so there's no actual beam search being done with beam size 1)

Reply:
Maybe we can add some flag to allow to keep beam search logic with beam size of 1? Useful for the tests + might be quite useful for debugging issues as well.

Reply:
The beam search logic doesn't work for size 1, but I can look info it in a separate pr